x/sequencer/keeper/msg_server_decrease_bond.go

```diff
@@ -0,0 +1,49 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension/v3/x/sequencer/types"
+)
+
+// DecreaseBond implements types.MsgServer.
+func (k msgServer) DecreaseBond(goCtx context.Context, msg *types.MsgDecreaseBond) (*types.MsgDecreaseBondResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	sequencer, err := k.bondUpdateAllowed(ctx, msg.GetCreator())
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if the sequencer has enough bond to decrease
```

Comment:
should be checked against `effectiveBond` instead of `sequencer.Tokens`.
also both checks (`effectiveBond > 0`, `effectiveBond> minBond`) can be combined into oneliner for readability